third_party/move/move-vm/runtime/src/storage/loader/lazy.rs

```diff
@@ -37,16 +36,13 @@ where
     fn charge_module(
         &self,
         gas_meter: &mut impl GasMeter,
-        traversal_context: &mut TraversalContext,
+        traversal_context: &mut dyn ModuleTraversalContext,
         module_id: &ModuleId,
     ) -> PartialVMResult<()> {
-        let module_id = traversal_context
```

Comment:
what was this strange line? (and how did I miss it)

Reply:
This juts allocates the id? The issue is that current context requires `&'a` but we do not have ids that live long enough so if you insert, you need to allocate it to extend the lifetime. This is now part of `visit_if_not_special_module_id ` implementation.

Reply:
another thing I found later lol